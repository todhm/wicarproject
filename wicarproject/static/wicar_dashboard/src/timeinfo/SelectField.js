import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import {connect } from 'react-redux'
import {changeAvailabilityTime} from './action'



const SelectField=(props)=>{

        const{optionList,defaultLabel,className,name,index,changeAvailability,value,label,onChange} = props
        return(
                <span className={className}>
                    <select name={name} className="required" value = {value} onChange={onChange} >
                        {optionList.map((option,index)=>(
                             <option value={option.value} key={option.value}>{option.label}</option>
                        ))}
                    </select>
                    <span className="text">{label}</span>
                </span>
        )
    }




export default SelectField;
