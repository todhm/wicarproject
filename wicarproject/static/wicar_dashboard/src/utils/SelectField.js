import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import {TextField,MenuItem} from '@material-ui/core';
import Select from '@material-ui/core/Select';


const  SelectField=(props)=>{
    const {name,label,value,onChange,error,defaultLabel,optionList,labelEqual,noDefault} = props
    return(
        <TextField
            select
            fullWidth={true}
            name={name}
            label={label}
            value={value}
            onChange={onChange}
            helperText={error}
            error={error?true:false}
            margin="normal"
            >
        {noDefault?
            null:
            <option value="">{defaultLabel}</option>
        }
          {optionList.map((option)=>(
                  labelEqual?
                  <option value={option.codeName} key={option.codeName}>{option.codeName}</option>
                  :<option value={option.value} key={option.value}>{option.label}</option>
          ))}
      </TextField>
    )
}


export default SelectField;
