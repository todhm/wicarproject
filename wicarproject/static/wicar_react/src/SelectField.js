import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import { Link } from 'react-router-dom'
import { Route } from 'react-router-dom'
import PropTypes from 'prop-types'


class SelectField extends Component{

    render(){
        return(
            <div className="grid-item grid-item--3">
                <label htmlFor="js-makesInput">{this.props.label}</label>
                <span className="styled-select-container styled-select-container--fluid">
                    <select id="js-makesInput" name={this.props.name} className="required" value = {this.props.value}  onChange={this.props.onChange} >
                        <option value="">{this.props.defaultLabel}</option>
                        {this.props.optionList.map((option)=>(
                             <option value={option.codeName} key={option.codeName}>{option.codeName}</option>
                        ))}
                    </select>
                    {this.props.value===""?
                        <span className="text brand-select">{this.props.defaultLabel}</span>:
                        <span className="text">{this.props.value}</span>
                        }
                        <span className="errorMessage-text js-locationErrorText">{this.props.error}</span>

                </span>
            </div>
        )
    }
}


export default SelectField;
