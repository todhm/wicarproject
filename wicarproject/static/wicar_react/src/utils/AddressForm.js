import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import { Link } from 'react-router-dom'
import { Route } from 'react-router-dom'
import PropTypes from 'prop-types'
import * as RegisterApi from './RegisterApi'
import Select from 'react-select';
import { debounce,throttle } from 'lodash';
import axios from 'axios';



class AddressForm extends Component {

    static contextTypes = {
       router: PropTypes.object
     }

      state={
          options:[]
      }

      componentWillMount(){
          const {value,valueObj,label,error} = this.props;
          this.setState({value,valueObj})
      }




    getAddressOpts = debounce((address) => {
        if (!address || address.length < 2) {
            this.setState((prevState)=>{options:[]})
        }
        else{
            address =address.replace("/","")
            let addressObj ={}
            addressObj['address'] = address;
            let url= '/api/get_address_info';
            axios({url:url,
                          method:'post',
                          data:addressObj,
                          withCredentials:true
                      })
            .then((response) => {
                if(response.data){
                    const options= response.data.map((data)=>{
                    var return_data = {}
                    return_data['value'] = data.address
                    return_data['label'] = data.address
                    return_data['addressObj'] = data
                    return return_data
                    })
                  this.setState({options})
              }}).catch((error)=>{this.setState({options:[]})})
          }
      },500,{'leading':false,'trailing':true})

    onInputChange=(option)=>{this.getAddressOpts(option)}

    onChange = (val)=>{
        const value = val ===null ? '' : val
        this.setState((prevState)=>({value:value,
                                    valueObj:value.addressObj}))

    }


    handleChange(event) {
        this.setState({value: event.target.value})
    }


    moveCaretAtEnd(e) {
      var temp_value = e.target.value
      e.target.value = ''
      e.target.value = temp_value
    }


    render(){
        const {onChange,value,valueObj,label,error} = this.props;
        return(
            <div className="section">
                <label className="section-heading">{label} </label>
                <Select
                    name="address"
                    ref={(ref) => { this.select = ref; }}
                    searchPromptText="주소나 지명을 입력해주세요"
                    loadingPlaceholder=""
                    autoload={true}
                    autoFocus={true}
                    openOnFocus={true}
                    options={this.state.options||[]}
                    onChange={onChange}
                    value={value}
                    onInputChange={this.onInputChange.bind(this)}
                    backspaceRemoves={true}
                    />
                <span className="errorMessage-text js-locationErrorText">{error}</span>
            </div>
        )
    }
}


export default AddressForm;
