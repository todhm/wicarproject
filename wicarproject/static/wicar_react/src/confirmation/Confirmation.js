import React, { Component } from 'react';
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'
import * as RegisterApi from '../utils/RegisterApi'



class Confirmation extends Component{

    static contextTypes = {
       router: PropTypes.object
     }


    handleSubmit=(e)=>{
        e.preventDefault()
        var pathList = this.context.router.history.location.pathname.split("/",-1)
        var carid = (pathList.length>=3) ? pathList[pathList.length-1]:"";
        RegisterApi.activateCar(carid).then((data)=>{
            if(data.message==="success")
                window.location.href="/get_car_list"
        });
    }

    backPage= ()=>{
        var pathList = this.context.router.history.location.pathname.split("/",-1)
        var carid = (pathList.length>=3) ? pathList[pathList.length-1]:"";
        return "/register_pic" + "/"+carid;
    }
    render(){
        return(
            <form id="js-listingPublicationForm" onSubmit={this.handleSubmit}>

              <h5>자동차 등록 완료하기</h5>

              <p>해당 버튼을 클릭하면 자동차 등록이 완료됩니다.</p>

              <div className="buttonWrapper">
                  <Link to ={this.backPage()} id="back" className="button">이전페이지</Link>
                  <button className="submit button button--purple" type="submit">
                      완료
                  </button>
              </div>
              </form>
        )
    }

}



export default Confirmation;
