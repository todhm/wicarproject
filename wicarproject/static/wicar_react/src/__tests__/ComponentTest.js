import React from 'react';
import {shallow,mount} from 'enzyme';
import SelectField from '../SelectField'
import AddressForm from '../AddressForm'
import Confirmation from '../confirmation/Confirmation'
import { Link } from 'react-router-dom'
import	renderer	from	'react-test-renderer';
import { MemoryRouter } from 'react-router'

let optionList=[
    {codeName:"test3"},
    {codeName:"test4"}
]
test('SelectField	renders	properly', ()	=>	{
    const   wrapper	=	shallow(<SelectField
                                    label="테스트"
                                    defaultLabel="테스트라벨"
                                    value="test1"
                                    error=""
                                    optionList={optionList}
                                  />);
    const   element	=	wrapper.find('span');
    expect(element.length).toBe(3);
    expect(element.get(0).props.className).toBe("styled-select-container styled-select-container--fluid");
    expect(element.get(1).props.children).toBe('test1');
});



test('UsersList	renders	a	snapshot	properly',	()	=>	{
    const	tree	=	renderer.create(
                                <SelectField
                                    label="테스트"
                                    defaultLabel="테스트라벨"
                                    value="test1"
                                    error=""
                                    optionList={optionList}
                                  />
                            ).toJSON();
    expect(tree).toMatchSnapshot();
});

test('Confirmation	renders	properly', ()	=>	{
    const   wrapper	=	mount(
        <MemoryRouter>
            <Confirmation/>
        </MemoryRouter>);
    const   element	=	wrapper.find('button');
    expect(element.length).toBe(1);
    expect(element.get(0).props.className).toBe("submit button button--purple");
});
