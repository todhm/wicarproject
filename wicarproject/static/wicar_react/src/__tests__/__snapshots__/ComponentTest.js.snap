// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList	renders	a	snapshot	properly 1`] = `
<div
  className="grid-item grid-item--3"
>
  <label
    htmlFor="js-makesInput"
  >
    테스트
  </label>
  <span
    className="styled-select-container styled-select-container--fluid"
  >
    <select
      className="required"
      id="js-makesInput"
      name={undefined}
      onChange={undefined}
      value="test1"
    >
      <option
        value=""
      >
        테스트라벨
      </option>
      <option
        value="test3"
      >
        test3
      </option>
      <option
        value="test4"
      >
        test4
      </option>
    </select>
    <span
      className="text"
    >
      test1
    </span>
    <span
      className="errorMessage-text js-locationErrorText"
    >
      
    </span>
  </span>
</div>
`;
